Tutorial: How to capture events from Calendly webhooks (#6789)
PostHog/posthog.com: contents/tutorials/calendly-webhooks.md

---
title: How to capture events from Calendly webhooks
date: 2023-09-21
author: ["ian-vanagas"]
showTitle: true
sidebar: Docs
featuredImage: ../images/tutorials/banners/tutorial-14.png
tags: ['events']
---

> Webhooks enable you to send data from one platform to another when an event happens. This enables you to run whatever workflows and code to handle those events.

Reviewer: ```suggestion
Webhooks enable you to send data from one platform to another when an event happens. This enables you to run workflows and code to handle those events.
```

> To showcase the power of webhooks, we are going to capture Calendly meeting data into PostHog using Val Town, a platform for writing and running JavaScript functions in your browser. 

Reviewer: hyperlink to calendly and Val Town

> > **Note:** Calendly webhooks requires their professional plan.

Reviewer: ```suggestion
> **Note:** Calendly webhooks requires subscribing their professional plan.
```

## Set up your Val Town webhook handler

> Before we set anything up in Calendly, we set up our webhook handler function in [Val Town](https://val.town/). 

Reviewer: I think its missing a sentence on what the role of this function will be e.g.
```suggestion
Before we set anything up in Calendly, we set up our webhook handler function in [Val Town](https://val.town/). This function will handle incoming events from calendly etc.
```

> To get started, create a variable file in Val Town with your `phProjectAPIKey`. Get your project API key from your [project settings](https://app.posthog.com/project/settings), go to Val Town, create a function like `let phProjectAPIKey = ‘<ph_project_api_key>’` and press "Run". This saves the value for us to reference as `@me.phProjectAPIKey` later.

Reviewer: ```suggestion
To get started, we'll create a variable file in Val Town with your PostHog API key. 

You can get your project API key from your [project settings](https://app.posthog.com/project/settings). Then, go to Val Town, create a function with the code `let phProjectAPIKey = ‘<ph_project_api_key>’` and press "Run". This saves the value for us to reference as `@me.phProjectAPIKey` later.
```

![Project API key Val](../images/tutorials/calendly-webhooks/val.png)

Next, we create our function to capture a Calendly webhook. It takes a request, parses the JSON, gets the data we want, and then uses PostHog to capture it.

> > Val Town enables you to reference or fork functions other people wrote, like the `capturePostHogEvent` [function](https://www.val.town/v/ianvph.capturePostHogEvent) we can use here.

Reviewer: Instead of mention this, in your code, can we just add the code from "  @ianvph.capturePostHogEvent(", instead of referencing somewhere else? WIll be easier to follow

Altogether, this function looks like:

<iframe src="https://www.val.town/embed/ianvph.captureCalendlyWebhook" height="635" frameBorder="0" allowFullScreen></iframe>

You can fork this function to customize it to your needs. 

Once, done click "Save," set as unlisted, click the three dots, hover the "endpoints" options, and click "copy web endpoint." This is the URL we use for our Calendly webhook trigger. It is in the format `https://{username}-{valname}.web.val.run`.

## Setting up your Calendly webhook

> Once you upgrade your Calendly account, you can create a webhook through their API. To start, you need a personal access token which you can create on the [API webhooks integration page](https://calendly.com/integrations/api_webhooks). 

Reviewer: ```suggestion
Once you upgrade your Calendly account to <plan name>, you can create a webhook using their API. 

To start, you need a personal access token which you can create on the [API webhooks integration page](https://calendly.com/integrations/api_webhooks). 
```
